lib/strong_migrations/migration_helpers.rb

```diff
@@ -73,16 +75,86 @@ def remove_null_constraint_safely(table_name, column_name, name: nil)
       end
     end
 
+    def add_column_safely(table_name, column_name, type, **options)
+      ensure_postgresql(__method__)
+      ensure_not_in_transaction(__method__)
+
+      default = options.delete(:default)
+
+      if postgresql_version >= 110000 || default.nil?
+        add_column(table_name, column_name, options)
+      else
+        reversible do |dir|
+          dir.up do
+            transaction do
+              add_column(table_name, column_name, type, default: nil, **options) unless connection.column_exists?(table_name, column_name, type)
+              change_column_default(table_name, column_name, default)
+            end
+
+            default_after_type_cast = connection.type_cast(default)
+            backfill_column_safely(table_name, column_name, default_after_type_cast)
+
+            allow_null = options[:null]
+            add_null_constraint_safely(table_name, column_name) unless allow_null
+          end
+
+          dir.down do
+            remove_column(table_name, column_name)
+          end
+        end
+      end
+    end
+
+    def backfill_column_safely(table_name, column_name, value, batch_size: 1000)
+      ensure_not_in_transaction(__method__)
+
+      table = Arel::Table.new(table_name)
+      primary_key = connection.primary_key(table_name)
+
+      start_arel = table
+        .project(table[primary_key])
+        .where(table[column_name].not_eq(value))
+        .order(table[primary_key].asc)
+        .take(1)
+
+      result = connection.exec_query(start_arel.to_sql)
+      return if result.empty?
+
+      start_pk = result.first[primary_key]
+
+      loop do
+        finish_arel = table
+          .project(table[primary_key])
+          .where(table[primary_key].gteq(start_pk))
+          .order(table[primary_key].asc)
+          .skip(batch_size)
+          .take(1)
+
+        finish_result = connection.exec_query(finish_arel.to_sql).first
+
+        update_arel = Arel::UpdateManager.new
+          .table(table)
+          .set([[table[column_name], value]])
+          .where(table[primary_key].gteq(start_pk))
+
+        if finish_result
+          finish_pk = finish_result[primary_key]
+          update_arel = update_arel.where(table[primary_key].lt(finish_pk))
+          start_pk = finish_pk
+        end
+
+        safety_assured { execute(update_arel.to_sql) }
+
```

Comment:
Need to throttle - `sleep(0.01)` (may need a global option for this later since more powerful databases may need less throttling, but let's deal with that later)